dlc-oracle-test/src/test/scala/org/bitcoins/dlc/oracle/AttestationVerificationTest.scala

```diff
@@ -0,0 +1,327 @@
+    assert(
+      !OracleEvent.verifyAttestations(unsignedDigitDecompTlv,
+                                      Vector.empty,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(unsignedDigitDecompTlv,
+                                      validEnumAttestation,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(unsignedDigitDecompTlv,
+                                      invalidUnsignedDigitDecompAttestation,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(unsignedDigitDecompTlv,
+                                      invalidUnsignedDigitDecompAttestation1,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(
+        unsignedDigitDecompTlv,
+        validUnsignedDigitDecompAttestation.reverse,
+        signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(
+        unsignedDigitDecompTlv,
+        SchnorrDigitalSignature(
+          validUnsignedDigitDecompAttestation.head.rx,
+          FieldElement.fromHex(
+            "0001020304050607080910111213141516171819202122232425262728293031")) +: validUnsignedDigitDecompAttestation.tail,
+        signingVersion
+      ))
+  }
+
+  it should "validate valid signed digit decomp attestations" in {
+    assert(
+      OracleEvent.verifyAttestations(signedDigitDecompTlv,
+                                     validSignedDigitDecompAttestation,
+                                     signingVersion))
+  }
+
+  it should "not validate invalid signed digit decomp attestations" in {
+    assert(
+      !OracleEvent
+        .verifyAttestations(signedDigitDecompTlv, Vector.empty, signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(signedDigitDecompTlv,
+                                      validEnumAttestation,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(signedDigitDecompTlv,
+                                      invalidSignedDigitDecompAttestation,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(signedDigitDecompTlv,
+                                      invalidSignedDigitDecompAttestation1,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(signedDigitDecompTlv,
+                                      validSignedDigitDecompAttestation.reverse,
+                                      signingVersion))
+    assert(
+      !OracleEvent.verifyAttestations(
+        signedDigitDecompTlv,
+        SchnorrDigitalSignature(
+          validSignedDigitDecompAttestation.head.rx,
+          FieldElement.fromHex(
+            "0001020304050607080910111213141516171819202122232425262728293031")) +: validUnsignedDigitDecompAttestation.tail,
+        signingVersion
+      ))
+  }
+
```

Comment:
Might be worth adding a test that goes through the process of creating the signature and verifying it.